### ISSUE TITLE
copy and paste - hanging

### ISSUE THREAD

**Issue by tompkins-ct** [gaphor/gaphor#3256]
### Describe the bug
Copy and paste between diagrams on large models hangs for a long time. Using a model with ~8k-9k elements. Copying single or multiple items from one diagram to another causes Gaphor to pause (on Mac spinning wheel). Pasting occurs eventually, but only after a long delay with no function (~5-10 min). Time delay seems to be a function of how much is being copied to the new/different diagram.

### To Reproduce
1. Make a large model
2. copy some elements (cmd + c) on a diagram
3. open a different or new diagram 
4. Paste with (cmd + v)
5. wait...

 No errors found:
```
  Gaphor version:         2.25.0
  Operating System:       Darwin (23.4.0)
  Display:                GdkMacosDisplay
  Python version:         3.12.2
  GTK version:            4.14.2
  Adwaita version:        1.5.0
  GtkSourceView version:  5.12.0
  Cairo version:          1.18.0
  Pango version:          1.52.2
  PyGObject version:      3.48.2
  Pycairo version:        1.26.0
  
  
  No errors have been intercepted. We’re good.
```

### OS
- [ ] Linux (Please put in notes the specific distro)
- [X] macOS
- [ ] Windows

### Version
Version of Gaphor: 2.25.0

**Comment by amolenaar** [comment]
Hi,

I can reproduce this. It shows me a lot of messages on the console:

```
(gaphor:68330): Gtk-CRITICAL **: 08:25:21.393: gtk_widget_get_parent: assertion 'GTK_IS_WIDGET (widget)' failed

(gaphor:68330): Gtk-CRITICAL **: 08:25:21.393: gtk_widget_get_parent: assertion 'GTK_IS_WIDGET (widget)' failed

(gaphor:68330): Gtk-CRITICAL **: 08:25:21.393: gtk_widget_get_parent: assertion 'GTK_IS_WIDGET (widget)' failed
```

There is an [issue in GTK](https://gitlab.gnome.org/GNOME/gtk/-/issues/6642) at the moment that causes the UI to not update. Normally it looks like it's refreshing after some other events (e.g. mouse movement). This may have something to do with this issue.

**Comment by tompkins-ct** [comment]
@amolenaar  Can confirm, this is fixed in the new release 2.25.1
